File: for-local-test/python/231.power-of-two.py
class Solution:
    def isPowerOfTwo(self, n: int) -> bool:
        if n <= 0:
            return False
        if n == 1:
            return True
        power_of_two = 1
        while power_of_two < n:
            power_of_two *= 2
            if power_of_two == n:
                return True
        return False
    
if __name__ == "__main__":
    solution = Solution()
    n = 16
    print(solution.isPowerOfTwo(n))
    

File: for-local-test/231.power-of-two.java

class Solution {
    public boolean isPowerOfTwo(int n) {
        if (n <= 0) return false;
        if(n == 1) return true;
        long powerOfTwo = 1;
        while (powerOfTwo < n){
            powerOfTwo *= 2;
            if(powerOfTwo == n) return true;
        }
        return false;
    }
    //
    public boolean isPowerOfTwoBitwise(int n) {
        if (n <= 0) return false;
        return (n & (n - 1)) == 0;
    }

    public static void main(String[] args) {
        Solution solution = new Solution();
        System.out.println(solution.isPowerOfTwo(1)); // true
        System.out.println(solution.isPowerOfTwo(16)); // true
        System.out.println(solution.isPowerOfTwo(3)); // false
        System.out.println(solution.isPowerOfTwoBitwise(1)); // true
        System.out.println(solution.isPowerOfTwoBitwise(16)); // true
        System.out.println(solution.isPowerOfTwoBitwise(3)); // false
    }
}

